### PR TITLE
fix: enforce full line to match when reference regex has the form ^...$

### PR DESCRIPTION
**Pull request by moormaster**

### CODE REVIEW

**Comment by moormaster** [comment]
Please also check if translations are ok - I don't speak sv nor pt-br :)

File: _data/localization.json [hunk 1]
```diff
@@ -89,7 +89,12 @@
       "de": "✓ gefunden",
       "sv": "✓ matchade",
       "pt-br": "✓ corresponde"
-      
+    },
+    "matchedIncompletely": {
+      "en": "(✓) matched, but whole line should match",
```
Comment (niccokunzmann) [inline _data/localization.json:94]: Ah, cool the checkmark is in ()..

File: en/05-02.html [hunk 2]
```diff
@@ -5,8 +5,8 @@
   The following expression shall find the numbers 20 to 29 but it has mistakes.
   Would you like to change it so it matches only the number from 20 to 29?
 </p>
-<div class="playfield">
-  <input type="text" value="[20-29]" class="regex" reference="2[0-9]">
+<div class="playfield verbose">
+  <input type="text" value="[20-29]" class="regex" reference="^2[0-9]$">
```
Comment (niccokunzmann) [inline en/05-02.html:9]: I cannot remember where  `^` and `$` are introduced. Is is important to not use them before that. Are they introduced before?


Here, it seems we learn sets.

**Comment by niccokunzmann** [comment]
I think, I cannot check the translations. But if someone from that language finds the translations misleading, they can open a new PR. 

I only have this one question: https://github.com/CoderDojoPotsdam/regex-tutorial/pull/121#discussion_r1574474899

**Comment by niccokunzmann** [comment]
Thanks for  the PR!

File: en/05-02.html (hunk 2 above)
Comment (moormaster) [reply en/05-02.html:9]: The `^` and `$` checkmarks in the reference string are only used to mark the challenge as "the matching part of the user pattern has to be fully equal to the example string". The user is not enforced to use `^` or `$` to achieve this.

If you don't want that behaviour for a particular challenge then you can just use a reference pattern that does not start with `^` and does not end with `$` and you will get the "old" behaviour. 

As you can see in challenge 05-02: It is enough if the user provides a pattern that matches both digits of the numbers between 20 and 29 to satisfy the challenge even though the reference regex uses these marks.

**Comment by moormaster** [comment]
force pushed:
added a small fix that ensured the overall result to not be "green" until every incomple match was fixed

**Comment by niccokunzmann** [comment]
Thanks for all that work on the project! I will continue to help in that. Let me know if you need anything!

**Pull request merged**